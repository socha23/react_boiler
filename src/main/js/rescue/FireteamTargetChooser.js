import React from 'react'
import {connect} from 'react-redux'
import {withRouter} from 'react-router'

import TaggedObjectsList from '../tags/TaggedObjectsList'

const RescueTargetsChooser = ({
        selected = {},
        tags = [],
        onSelect = () => {}
    }) =>
        <TaggedObjectsList selected={selected} tags={tags} onSelect={onSelect}/>;

export default RescueTargetsChooser