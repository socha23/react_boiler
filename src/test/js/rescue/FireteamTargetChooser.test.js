import React from 'react'
import {testSnapshotWithProvider, expectElemWithProvider} from '../testUtils'

import FireteamTargetChooser from 'rescue/FireteamTargetChooser'

const T_1 = {
    id: "T_1",
    state: "INSIDE",
    areaName: "inside"
};
const T_2 = {
    id: "T_2",
    state: "INSIDE",
    areaName: "inside"
};
const T_3 = {
    id: "T_3",
    state: "MISSING",
    areaName: "inside"
};
const T_4 = {
    id: "T_4",
    state: "INSIDE",
    areaName: "inside"
};

const FT = {
    id: "FT",
    name: "fireteam",
    tagId: T_1.id,
    targetTagId: T_2.id
};

const A_1 = {
    id: "A_1",
    name: "artifact_1",
    priority: "P1_LOW",
    tagId: T_2.id
};
const A_2 = {
    id: "A_2",
    name: "artifact_2",
    priority: "P1_LOW",
    tagId: T_3.id
};



const STATE = {
        tags: {
            items: [T_1, T_2, T_3, T_4],
            itemsById: {T_1: T_1, T_2: T_2, T_3: T_3, T_4: T_4}
        },
        artifacts: {
            items: [A_1, A_2],
            itemsById: {A_1: A_1, A_2: A_2}
        },
        fireteams: {
            items: [FT],
            itemsById: {FT: FT}
        }
    };


it('renders correctly', testSnapshotWithProvider(<FireteamTargetChooser/>, STATE));

it('renders correctly when state empty', testSnapshotWithProvider(
    <FireteamTargetChooser/>, {
        tags: {},
        artifacts: {},
        fireteams: {}
    }
));

it('shows artifact inside', () => {
    expectElemWithProvider(
        <FireteamTargetChooser/>, STATE
    ).toIncludeText("artifact_1")
});

it('doesnt show artifact with missing tag', () => {
    expectElemWithProvider(
        <FireteamTargetChooser/>, STATE
    ).not.toIncludeText("artifact_2")
});
