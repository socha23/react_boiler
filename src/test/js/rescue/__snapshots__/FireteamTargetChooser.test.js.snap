// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <nav>
    <ul
      style={
        Object {
          "listStyleType": "none",
          "marginBottom": 0,
          "paddingLeft": 15,
        }
      }
    >
      <li
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "border": "1px solid #ddd",
            "borderBottom": "1px solid white",
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
            "cursor": "pointer",
            "display": "inline-block",
            "fontSize": 18,
            "padding": "10px 16px 6px 16px",
            "position": "relative",
            "top": 1,
          }
        }
      >
        <a>
          Muzealia
        </a>
      </li>
      <li
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
            "display": "inline-block",
            "fontSize": 18,
            "padding": "10px 16px 6px 16px",
            "position": "relative",
            "top": 1,
          }
        }
      >
        <a>
          Roty
        </a>
      </li>
      <li
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
            "display": "inline-block",
            "fontSize": 18,
            "padding": "10px 16px 6px 16px",
            "position": "relative",
            "top": 1,
          }
        }
      >
        <a>
          Nawigacja
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="panel panel-default"
  >
    <div
      className="panel-body"
      style={
        Object {
          "padding": 0,
        }
      }
    >
      <div
        style={
          Object {
            "height": 0,
            "overflowY": "auto",
          }
        }
      >
        <table
          className="table table-hover table-pointer table-noTopPadding"
        >
          <tbody>
            <tr
              className=""
              onClick={[Function]}
            >
              <td>
                <div
                  style={
                    Object {
                      "alignItems": "center",
                      "display": "flex",
                      "paddingBottom": 5,
                      "paddingTop": 5,
                    }
                  }
                >
                  <span
                    className="vocIcon iconWithName"
                    title="Niski"
                  >
                    
                    <span
                      className="badge"
                    >
                      1
                    </span>
                  </span>
                  <div
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <span
                      style={
                        Object {
                          "fontSize": 20,
                        }
                      }
                    >
                      artifact_1
                    </span>
                    <div>
                      <small
                        style={
                          Object {
                            "color": "#AAA",
                            "fontSize": 16,
                          }
                        }
                      >
                        inside
                      </small>
                      <span
                        className="label label-success"
                        style={
                          Object {
                            "fontSize": 16,
                            "marginLeft": 10,
                          }
                        }
                      >
                        fireteam
                      </span>
                    </div>
                  </div>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly when state empty 1`] = `
<div>
  <nav>
    <ul
      style={
        Object {
          "listStyleType": "none",
          "marginBottom": 0,
          "paddingLeft": 15,
        }
      }
    >
      <li
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "border": "1px solid #ddd",
            "borderBottom": "1px solid white",
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
            "cursor": "pointer",
            "display": "inline-block",
            "fontSize": 18,
            "padding": "10px 16px 6px 16px",
            "position": "relative",
            "top": 1,
          }
        }
      >
        <a>
          Muzealia
        </a>
      </li>
      <li
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
            "display": "inline-block",
            "fontSize": 18,
            "padding": "10px 16px 6px 16px",
            "position": "relative",
            "top": 1,
          }
        }
      >
        <a>
          Roty
        </a>
      </li>
      <li
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
            "display": "inline-block",
            "fontSize": 18,
            "padding": "10px 16px 6px 16px",
            "position": "relative",
            "top": 1,
          }
        }
      >
        <a>
          Nawigacja
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="panel panel-default"
  >
    <div
      className="panel-body"
      style={
        Object {
          "padding": 0,
        }
      }
    >
      <div
        style={
          Object {
            "height": 0,
            "overflowY": "auto",
          }
        }
      >
        <table
          className="table table-hover table-pointer table-noTopPadding"
        >
          <tbody />
        </table>
      </div>
    </div>
  </div>
</div>
`;
